### PR TITLE
feat: update deposit service integration into getDepositAddress metho…

### PR DESCRIPTION
**Pull request by canhtrinh**
…d with new object param

### CODE REVIEW

File: src/libs/AxelarAssetTransfer.ts [hunk 1]
```diff
@@ -176,23 +188,30 @@ export class AxelarAssetTransfer {
...
+   * @param {string}  requestParams.toChain - Destination chain identifier eg: avalanche, moonbeam ethereum-2, terra-2 ...
+   * @param {string}  requestParams.destinationAddress - Address where the asset should be transferred to on the destination chain
+   * @param {string}  requestParams.asset - Asset denomination eg: uausdc, uaxl ... If the asset specific is native cxy (e.g. ETH, AVAX, etc), the ERC20 version of the asset will appear on the dest chain
+   * @param {Object}  requestParams.options
+   * @param {string}  requestParams.options._traceId
+   * @param {boolean} requestParams.options.shouldUnwrapIntoNative - when sending wrapped native asset back to its home chain (e.g. WETH back to Ethereum), specify "true" to receive native ETH; otherwise will received ERC20 version
+   * @param {string}  requestParams.options.refundAddress - recipient where funds can be refunded if wrong ERC20 asset is deposited; ONLY AVAILABLE FOR WRAP/UNWRAP SERVICE
+  */
+  public async getDepositAddress(requestParamsOrFromChain: GetDepositAddressParams | string, _toChain?: string, _destinationAddress?: string, _asset?: string, _options?: any): Promise<string> {
```
Comment (vladwulf) [inline src/libs/AxelarAssetTransfer.ts:201]: This line is too long, let's verify that prettier/eslint is enabled

File: src/libs/AxelarAssetTransfer.ts [hunk 2]
```diff
@@ -208,6 +227,39 @@ export class AxelarAssetTransfer {
     if (!isDestinationAddressValid)
       throw new Error(`Invalid destination address for chain ${toChain}`);
 
+    const chainList: ChainInfo[] = await loadChains({ environment: this.environment });
+
+    let srcChainInfo: ChainInfo = chainList.find(
```
Comment (vladwulf) [inline src/libs/AxelarAssetTransfer.ts:232]: redundant casting

Comment (vladwulf) [reply src/libs/AxelarAssetTransfer.ts:232]: should also be const

File: src/libs/AxelarAssetTransfer.ts [hunk 3]
```diff
@@ -208,6 +227,39 @@ export class AxelarAssetTransfer {
...
+      (chainInfo) => chainInfo.id === fromChain.toLowerCase()
+    ) as ChainInfo;
+    if (!srcChainInfo) throw new Error("cannot find chain" + fromChain);
+    let destChainInfo: ChainInfo = chainList.find(
```
Comment (vladwulf) [inline src/libs/AxelarAssetTransfer.ts:236]: same as above

File: src/libs/AxelarAssetTransfer.ts [hunk 4]
```diff
@@ -208,6 +227,39 @@ export class AxelarAssetTransfer {
...
+      (chainInfo) => chainInfo.id === toChain.toLowerCase()
+    ) as ChainInfo;
+    if (!destChainInfo) throw new Error("cannot find chain" + toChain);
+
+    /**if user has selected native cxy, e.g. ETH, AVAX, etc, assume it is to be wrapped into ERC20 on dest chain */
+    if (isNativeToken(srcChainInfo.chainName.toLowerCase() as EvmChain, asset as GasToken)) {
```
[the remark this answers is not in the document]
Comment (canhtrinh) [reply src/libs/AxelarAssetTransfer.ts:242]: good point, ok will change the casting here to string instead

**Pull request merged**